test/extended/olm/olmv1.go

```diff
@@ -142,28 +147,82 @@ var _ = g.Describe("[sig-olmv1] OLMv1 operator installation", func() {
-			}
-			if !meta.IsStatusConditionPresentAndEqual(conditions, "Progressing", metav1.ConditionFalse) {
-				return false, nil
-			}
-			if !meta.IsStatusConditionPresentAndEqual(conditions, "Installed", metav1.ConditionTrue) {
-				return false, nil
-			}
-			return true, nil
-		})
+		err = wait.PollUntilContextTimeout(ctx, time.Second, 5*time.Minute, true, func(ctx context.Context) (bool, error) { return ApplyClusterExtensionAndWait(oc, "install-test-ce") })
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("ensuring the cluster is upgradeable when no olm.maxopenshiftversion is specified")
+		var conditions []metav1.Condition
+		output, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("olm", "cluster", "-o=jsonpath={.status.conditions}").Output()
 		o.Expect(err).NotTo(o.HaveOccurred())
+		err = json.Unmarshal([]byte(output), &conditions)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(meta.IsStatusConditionTrue(conditions, typeIncompatibelOperatorsUpgradeable)).To(o.BeTrue())
+	})
+
+	g.It("should block cluster upgrades if an incompatible operator is installed", func(ctx g.SpecContext) {
+		const (
+			packageName = "elasticsearch-operator"
+			version     = "5.8.13"
+		)
+		// Check for tech preview, if this is not tech preview, bail
+		if !exutil.IsTechPreviewNoUpgrade(ctx, oc.AdminConfigClient()) {
+			g.Skip("Test only runs in tech-preview")
+		}
+
+		ns := oc.Namespace()
+		g.By(fmt.Sprintf("Updating the namespace to: %q", ns))
+		newCeFile = ceFile + "." + ns
+		b, err := os.ReadFile(ceFile)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		s := overwriteCeFile(b, ns, packageName, version)
+		err = os.WriteFile(newCeFile, []byte(s), 0666)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("applying the necessary resources")
+		err = oc.AsAdmin().WithoutNamespace().Run("apply").Args("-f", newCeFile).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("waiting for the ClusterExtention to be installed")
+		err = wait.PollUntilContextTimeout(ctx, time.Second, 5*time.Minute, true, func(ctx context.Context) (bool, error) { return ApplyClusterExtensionAndWait(oc, "install-test-ce") })
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("ensuring the cluster is upgradeable when no olm.maxopenshiftversion is specified")
+		var conditions []metav1.Condition
+		output, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("olm", "cluster", "-o=jsonpath={.status.conditions}").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = json.Unmarshal([]byte(output), &conditions)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(meta.IsStatusConditionTrue(conditions, typeIncompatibelOperatorsUpgradeable)).To(o.BeFalse())
 	})
 })
+
+func overwriteCeFile(b []byte, args ...string) string {
+	s := string(b)
+	s = strings.ReplaceAll(s, "{REPLACE}", args[0])
+	s = strings.ReplaceAll(s, "{PACKAGENAME}", args[1])
+	s = strings.ReplaceAll(s, "{VERSION}", args[2])
+	return s
+}
```

Comment:
I would this this function would actually do the read and write, based on its name.
The return value would be the new name of the file, plus an error...

Reply:
Maybe just a rename to something like `updateCe()`?